cpp/src/arrow/python/python_to_arrow.cc

```diff
@@ -1171,6 +1171,12 @@ Status GetConverterFlat(const std::shared_ptr<DataType>& type, bool strict_conve
 
 Status GetConverter(const std::shared_ptr<DataType>& type, bool from_pandas,
                     bool strict_conversions, std::unique_ptr<SeqConverter>* out) {
+  if (from_pandas) {
+    // ARROW-842: If pandas is not installed then null checks will be less
+    // comprehensive, but that is okay.
+    internal::InitPandasStaticData();
```

Comment:
Hmm... I think we're doing more general PyArrow initialization somewhere else. We should certainly not trust that only this code path will need the Pandas data.

Reply:
I see. Well, the problem with putting the pandas initialization elsewhere is that we've already arranged for pandas to only be imported when it's needed, so if we did `import pandas` always when doing `import pyarrow` we'd be breaking that contract. I suggest we address this as follow up as needed